plugins/catalog-backend-module-azure/src/processors/AzureDevOpsDiscoveryProcessor.test.ts

```diff
@@ -59,6 +61,20 @@ describe('AzureDevOpsDiscoveryProcessor', () => {
         project: 'engineering',
         repo: 'backstage',
         catalogPath: '/src/*/catalog.yaml',
+        branch: '',
+      });
+
+      expect(
+        parseUrl(
+          'https://azuredevops.mycompany.com/spotify/engineering/_git/backstage?path=/src/*/catalog.yaml&version=GBdevelopment',
+        ),
+      ).toEqual({
+        baseUrl: 'https://azuredevops.mycompany.com',
+        org: 'spotify',
+        project: 'engineering',
+        repo: 'backstage',
+        catalogPath: '/src/*/catalog.yaml',
+        branch: 'development',
```

Comment:
We should also test branches with a slash for example: `topic/catalog-info`

Reply:
Thank you for the feedback, that makes a lot of sense.

I've changed the tests to use a slash in the branch name. I don't think it's necessary to test with branches with no slash if we have this. Happy to include those tests though if you disagree.

I also ran some manual tests, both directly against the DevOps API, and with a local instance of Backstage, and I can confirm that using slashes in the branch name works fine. I found that it takes some time (up to an hour in my experience) for a branch that is added to the search index in Azure DevOps to become searchable. The API does not give any indication that the branch is being indexed and just returns no results, which created a bit of confusion for me, so I also added a note to the docs mentioning this.